src/ng/compile.js

```diff
@@ -2170,6 +2172,20 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     }
 
 
+    function downloadTemplateDependencies(directive) {
+      if (directive.directives) {
+        directive.directives.forEach(function(name) {
+          $injector.get(name + Suffix).forEach(function(dependency) {
+            downloadTemplateDependencies(dependency);
```

Comment:
traversing the graph without knowing what was already traversed, can get quite expensive fast